The HPX build fails
```
/home/runner/_work/kokkos/kokkos/kokkos/core/src/HPX/Kokkos_HPX.hpp:94:19: error: no matching function for call to 'min' [clang-diagnostic-error]
   94 |   const T end   = (std::min)(begin + chunk_size, max);
      |                   ^~~~~~~~~~
/home/runner/_work/kokkos/kokkos/kokkos/core/src/HPX/Kokkos_HPX.hpp:956:20: note: in instantiation of function template specialization 'Kokkos::Impl::get_chunk_range<short>' requested here
  956 |     const auto r = get_chunk_range(i_chunk, m_policy.begin(),
      |                    ^
/home/runner/_work/kokkos/kokkos/kokkos/core/src/HPX/Kokkos_HPX.hpp:327:58: note: in instantiation of member function 'Kokkos::Impl::ParallelFor<(lambda at /home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:478:7), Kokkos::RangePolicy<Kokkos::Experimental::HPX, Kokkos::IndexType<short>>>::execute_range' requested here
  327 |                            {[functor](Index i) { functor.execute_range(i); }},
      |                                                          ^
/home/runner/_work/kokkos/kokkos/kokkos/core/src/HPX/Kokkos_HPX.hpp:970:22: note: in instantiation of function template specialization 'Kokkos::Experimental::HPX::impl_bulk_plain<Kokkos::Impl::ParallelFor<(lambda at /home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:478:7), Kokkos::RangePolicy<Kokkos::Experimental::HPX, Kokkos::IndexType<short>>>, short>' requested here
  970 |     m_policy.space().impl_bulk_plain(false, is_light_weight_policy<Policy>(),
      |                      ^
/home/runner/_work/kokkos/kokkos/kokkos/core/src/Kokkos_Parallel.hpp:144:11: note: in instantiation of member function 'Kokkos::Impl::ParallelFor<(lambda at /home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:478:7), Kokkos::RangePolicy<Kokkos::Experimental::HPX, Kokkos::IndexType<short>>>::execute' requested here
  144 |   closure.execute();
      |           ^
/home/runner/_work/kokkos/kokkos/kokkos/core/src/Kokkos_Parallel.hpp:162:11: note: in instantiation of function template specialization 'Kokkos::parallel_for<char[1], Kokkos::RangePolicy<Kokkos::Experimental::HPX, Kokkos::IndexType<short>>, (lambda at /home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:478:7)>' requested here
  162 |   Kokkos::parallel_for("", policy, functor);
      |           ^
/home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:475:11: note: in instantiation of function template specialization 'Kokkos::parallel_for<Kokkos::RangePolicy<Kokkos::Experimental::HPX, Kokkos::IndexType<short>>, (lambda at /home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:478:7)>' requested here
  475 |   Kokkos::parallel_for(
      |           ^
/home/runner/_work/kokkos/kokkos/kokkos/core/unit_test/TestRange.hpp:498:5: note: in instantiation of function template specialization 'Test::test_small_index_type<short>' requested here
  498 |     test_small_index_type<short>();
      |     ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/algorithmfwd.h:427:5: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('int' vs. 'short')
  427 |     min(const _Tp&, const _Tp&);
      |     ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/stl_algo.h:5774:5: note: candidate template ignored: could not match 'initializer_list<_Tp>' against 'int'
 5774 |     min(initializer_list<_Tp> __l, _Compare __comp)
      |     ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/stl_algo.h:5764:5: note: candidate function template not viable: requires single argument '__l', but 2 arguments were provided
 5764 |     min(initializer_list<_Tp> __l)
      |     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-redhat-linux/15/../../../../include/c++/15/bits/algorithmfwd.h:432:5: note: candidate function template not viable: requires 3 arguments, but 2 were provided
  432 |     min(const _Tp&, const _Tp&, _Compare);
      |     ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

--- a/core/src/HPX/Kokkos_HPX.hpp
+++ b/core/src/HPX/Kokkos_HPX.hpp
@@ -91,7 +91,7 @@ template <typename T>
 constexpr hpx_range<T> get_chunk_range(const T i_chunk, const T offset,
                                        const T chunk_size, const T max) {
   const T begin = offset + i_chunk * chunk_size;
-  const T end   = (std::min)(begin + chunk_size, max);
+  const T end   = std::min(static_cast<T>(begin + chunk_size), max);
   return {begin, end};
 }
 

